src/plugins/intel_gna/src/backend/gna_limitations.hpp

```diff
@@ -284,178 +153,157 @@ class AbstractValidator {
+    bool UseOnly16BitConvolutionWeights() const;
+
+    inline bool IsCropAffinedOffset(size_t numberOfElements) const {
+        const auto cropOffset = numberOfElements * kBytesPerCropElement;
+        return (ALIGN64(cropOffset) != cropOffset);
+    }
+
+    inline size_t GetMemoryAlignment() const {
+        OPENVINO_ASSERT(GetCompileTarget() != target::DeviceVersion::NotSet);
+        return m_mem_alignment;
+    }
+
+    inline std::shared_ptr<cnn2d::AbstractValidator> GetCnnValidator() const {
+        return m_cnn_validator;
+    }
 
-public:
-    Validator_35() = default;
-
-    bool ValidateCnn2D(const std::string& name,
-                       const uint32_t inHeight,
-                       const uint32_t inWidth,
-                       const uint32_t inChannels,
-                       const uint32_t kH,
-                       const uint32_t kW,
-                       const uint32_t kN,
-                       const uint32_t strideH,
-                       const uint32_t strideW,
-                       const uint32_t dilationH,
-                       const uint32_t dilationW,
-                       OvGnaType inPrecision,
-                       bool exception = true) const override;
-
-    bool ValidatePooling2D(const std::string& name,
-                           const uint32_t windowH,
-                           const uint32_t windowW,
-                           const uint32_t strideH,
-                           const uint32_t strideW,
-                           bool exception = true) const override;
-
-    bool ValidateInputPadding(const std::string& name,
-                              const uint32_t pad_h_begin,
-                              const uint32_t pad_h_end,
-                              const uint32_t pad_w_begin,
-                              const uint32_t pad_w_end,
-                              const uint32_t kernel_h,
-                              const uint32_t kernel_w,
-                              const bool throwOnError = true) const override;
-
-    bool ShouldUseOnlyConv2DGnaIface() const override;
-
-    bool ValidateCnn1D(const std::string& name,
-                       const uint32_t inHeight,
-                       const uint32_t inWidth,
-                       const uint32_t inChannels,
-                       const uint32_t kH,
-                       const uint32_t kW,
-                       const uint32_t kN,
-                       const uint32_t strideH,
-                       const uint32_t strideW,
-                       const uint32_t dilationH,
-                       const uint32_t dilationW,
-                       OvGnaType inPrecision,
-                       bool exception = true) const override;
+private:
+    Limitations() {}
+    Limitations(const Limitations&) = delete;
+    Limitations& operator=(const Limitations&) = delete;
+
+    inline const target::DeviceVersion& GetCompileTarget() const {
```

Comment:
Why do we need additional function for getting compile target? Not so long ago the code was refactored to remove all duplicates of target manipulation functions/variables, so this should not be added again here.

Reply:
This is a private method of the Limitations class used in the Init function for validating the compile target. The reason m_compile_target cannot be used directly is that Init is a static type.

Reply:
as confirmed offline removig compile target Limitations fields